APPS-1366 dxcint test GHA

JanProvaznik opened the pull request.

Gvaihir committed:
dxcint
(fix) Terraform.py - no chdir in threads.

JanProvaznik committed:
(fix) test failures due to duplicate builds

JanProvaznik committed:
dxcint

(feat) GHA for testing dxcint - initial

JanProvaznik committed:
dxcint

(fix) dxcint tests

JanProvaznik committed:
Merge branch 'APPS-1389-dxCint-test-fails' into APPS-1366-dxcint-test-GHA

JanProvaznik committed:
dxcint

(fix) dxcint -locked in RegisteredTest compilation

Gvaihir committed:
Merge branch 'develop' into APPS-1366-dxcint-test-GHA

Gvaihir committed:
APPS-1366 dxcint test GHA
(fix) RegisteredTest.py - removed `-locked`. See comments to #380.
(fix) mock_1_input.json - applied Jan's suggestions

Gvaihir commented on .github/workflows/dxcint.yml:
```diff
@@ -0,0 +1,69 @@
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Install dxpy and other dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y apt-transport-https wget git openssh-server tree python3 python3-pip python3-venv
+          cwltool=`jq -r '.execDepends.cwl[0].name' scripts/bundled_dependencies.json`
+          pip3 install --upgrade pip setuptools pyOpenSSL pytest
+          pip3 install termcolor dxpy==0.323.0 $cwltool
+          # install dxda
+          dxda_version=`jq -r '.dxda' scripts/bundled_dependencies.json`
+          sudo wget -O/usr/local/bin/dx-download-agent https://github.com/dnanexus/dxda/releases/download/${dxda_version}/dx-download-agent-linux
+          sudo chmod +x /usr/local/bin/dx-download-agent
```
`dxcint` is a poetry package, where you have `poetry.lock` file with all dependency versions pinned - pretty handy and reproducible. Could you rewrite this bit with something like I did for [Scientific Product CI/CD](https://github.com/dnanexus/pseudopod/blob/ab0ba32a89216b18b02ebbcf3b4f41594ac52bd8/.github/workflows/sciprodbuild_unit_tests.yml#L44-L51)?

Also, not sure there's a need for `dx-download-agent`
The rest is perfect

Gvaihir reviewed (changes requested):
Good job! Just rewrite install for the `dxcint`

JanProvaznik committed:
dxcint

(fix) dxcint.yml - simplify installation with poetry
(fix) test_RegisteredTest.py - update input name

JanProvaznik committed:
minor

JanProvaznik committed:
minor

JanProvaznik committed:
minor

Gvaihir committed:
APPS-1366 dxcint test GHA
(fix) dxcint.yml - minor renaming

Gvaihir commented on .github/workflows/dxcint.yml:
```diff
@@ -0,0 +1,68 @@
+          java-version: 11
+
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
```
Nice, someone's done that, very useful

Gvaihir reviewed (approved):
Good job!
